scripts/generate-sitemap.js

```diff
@@ -3,23 +3,67 @@
 const fs = require('fs')
 const { parseStringPromise, Builder } = require('xml2js')
 
+const weekMs = 7 * 24 * 60 * 60 * 1000
+const nowISO = new Date().toISOString()
+
+const getQuery = (chain) => `
+  query {
+    topTokens(pageSize: 100, page: 1, chain: ${chain}, orderBy: VOLUME) {
+      address
+    }
+  }
+`
+const chains = ['ETHEREUM', 'ARBITRUM', 'OPTIMISM', 'POLYGON', 'BASE', 'BNB', 'CELO']
+
 fs.readFile('./public/sitemap.xml', 'utf8', async (err, data) => {
+  const sitemapURLs = {}
   try {
     const sitemap = await parseStringPromise(data)
-
-    const lastmodDate = new Date().toISOString()
     if (sitemap.urlset.url) {
       sitemap.urlset.url.forEach((url) => {
-        url['$'].lastmod = lastmodDate
+        const lastMod = new Date(url['$'].lastmod).getTime()
+        if (lastMod < Date.now() - weekMs) {
+          url['$'].lastmod = nowISO
+        }
+        sitemapURLs[url['$']['loc']] = true
       })
     }
+
+    for (const chainName of chains) {
+      const response = await fetch('https://api.uniswap.org/v1/graphql', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Origin: 'https://app.uniswap.org',
+          'User-Agent':
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.5735.110 Safari/537.36',
+        },
+        body: JSON.stringify({ query: getQuery(chainName) }),
+      })
+      const tokensJSON = await response.json()
+      const tokenAddresses = tokensJSON.data.topTokens.map((token) => token.address.toLowerCase())
```

Comment:
do we want all 100 top tokens in the sitemap?  + if this array is changing on a frequent basis (say hourly), does this mean our sitemap is also changing hourly?

Reply:
i.e. top token #100 rn is Alien Milady Fumo lol

Reply:
i think we do, the sitemap is meant to be an index of as many pages as possible. and i think we do want the "token of the month" in the sitemap since more people will be searching for it while it's trending, and we want our TDP for that token to rank highly